modules/cloudfront.md

```diff
@@ -0,0 +1,220 @@
+    "trusted-signers": [],
+    "viewer-protocol-policy": "https-only",
+    "min-ttl": 86400,
+    "max-ttl": 86400,
+    "default-ttl": 86400,
+    "smooth-streaming": false,
+    "allowed-methods": [
+      "HEAD",
+      "GET",
+      "OPTIONS"
+    ],
+    "cached-methods": [
+      "HEAD",
+      "GET"
+    ]
+  },
+  "cache-behaviors": [
+    {
+      "target-origin-id": "S3-bucket-name-1",
+      "path-pattern": "*.png",
+      "forward-query-strings": false,
+      "forwarded-cookies": "none",
+      "forward-headers": [],
+      "trusted-signers": [],
+      "viewer-protocol-policy": "https-only",
+      "min-ttl": 31536000,
+      "max-ttl": 31536000,
+      "default-ttl": 31536000,
+      "smooth-streaming": false,
+      "allowed-methods": [
+        "HEAD",
+        "GET",
+        "OPTIONS"
+      ],
+      "cached-methods": [
+        "HEAD",
+        "GET",
+        "OPTIONS"
+      ]
+    }
+  ],
+  "comment": "Created with Cumulus",
+  "enabled": false
+}
+{% endhighlight %}
+
+It should be noted that Cumulus will create a new CloudFront distribution if you omit the `"id"` parameter the next time the configuration is synced. After creation, the configuration used to create the distribution will be updated with the id of the created distribution to prevent it from being created again on accident.
+
+There are some configuration options for web distributions that Cumulus does not handle because we do not use them at Lucid or do not want them managed by Cumulus at this time. These include:
+
+* `default_root_object`
+* `logging`
+* `price_class`
+* `viewer_certificate`
+* `restrictions`
+
+If you would like these added to Cumulus, please submit a pull request.
+
+
+Diffing and Syncing CloudFront
+------------------------------
+
+Cumulus's CloudFront module has the following usage:
+
+{% highlight bash %}
+cumulus cloudfront [diff|help|invalidate|list|migrate|sync] <asset>
+{% endhighlight %}
+
+Distributions can be diffed, listed, and synced (migration is covered in the [following section](#migration)). The three actions to the following:
+
+* `diff` - Shows the differences between the local definition and the AWS CloudFront configuration. If `<asset>` is specified, Cumulus will diff only the distribution defined in that file.
+* `list` - Lists the names of the files that contain distribution definitions
+* `sync` - Syncs local configuration with AWS. If `<asset>` is specified, Cumulus will sync only the distribution defined in the file with that name.
+* `invalidate` - Creates an invalidation based on the config specified by `<asset>`.  Information can be found in the [invalidation section](#invalidation)
+
+Migration
+---------
+
+If your environment is anything like ours, you have dozens of distributions, and would rather not write Cumulus configuration for them by hand. Luckily, Cumulus provides a `migrate` task that will pull down your web distributions and produce configuration for them.
+
+Invalidation
```

Comment:
Because invalidations are different than the day to day management of the distribution it may be nice to give an example command of executing an invalidation.